### ISSUE TITLE
VS 2015 CTP 6 crashes when "Debug in F# interactive" executed

### ISSUE THREAD

**Issue by vasily-kirichenko** [Microsoft/visualfsharp#291]
1. Add a script with the following content
```fsharp
open System
let dt = DateTime.Now
printfn "%A\n" 1
```
2. Set a breakpoint on `let dt = DateTime.Now` line (yes, on the screenshot the breakpoint is moved to the line below, but initially it was on `let dt..` line).
3. Execute "Debug in F# interactive" command from the context menu.
4. VS crashes with the following message:

![image](https://cloud.githubusercontent.com/assets/873919/6449772/41db7a00-c134-11e4-9ecd-7e475a88d797.png)

(it reads "VS stopped working")

**Comment by isaacabraham** [comment]
+1 - seeing this as well.

**Comment by latkin** [comment]
I just tried installing the latest F# Preview over VS 2015 Preview and do not repro.  I'll try installing on top of VS 2015 CTP6. Seems that there is surely an issue.

- What code are you highlighting (if any) when you choose to debug?
- What are the FSI settings? (Tools -> Options -> F# Tools -> F# Interactive)
- Is script debugging working otherwise and this code snippet is unique? Or does it not work at all?

**Comment by isaacabraham** [comment]
The code I highlighted was: -

    let x = 5
    x + 5

Options are: -
    64 bit = true
    options = "--optimize"
    shadow copy = true

**Comment by latkin** [comment]
We should add better UI to help with this, but FWIW you'll only get a good debug experience if you set `--optimize-`.  Regardless, that does not cause a crash for me.

Also curious - what are your debugging options?  It is critical that you do not enable "managed compatibility mode" for F# code in VS 2015+. We have found that there are a raft of issues with this legacy version of the debugger.

![image](https://cloud.githubusercontent.com/assets/5943573/6469146/e9be8974-c18e-11e4-96b0-4755e6e13bc6.png)

**Comment by vasily-kirichenko** [comment]
![image](https://cloud.githubusercontent.com/assets/873919/6470049/1b2a0a3e-c1f2-11e4-9600-c6ff66b4d5f9.png)

![image](https://cloud.githubusercontent.com/assets/873919/6470065/361a5dbc-c1f2-11e4-9dc2-230e42ffbcab.png)

I installed CTP6 as an upgrade for CTP5.

I enlarged the code to 

```fsharp
open System

let dt = DateTime.Now
printfn "%A\n" 1

[1..10] |> List.iter (printfn "%d")
System.Console.ReadKey() 
```
and VS crashed on it as well.

**Comment by isaacabraham** [comment]
BTW - I hadn't used CTP5 - I went straight from 4 (I think).

**Comment by latkin** [comment]
I just did a fresh install of CTP6 + latest F# preview and now I do repro.  Will take a look...

**Comment by latkin** [comment]
This is an issue with the Roslyn expression evaluators. Crash info should be in the event log.  Something must have changed between the VS 2015 bits I was using before and what's in CTP6.  F# sets a reg key that is supposed to have us fall back to the legacy EEs, but for some reason we are hitting trouble with a Roslyn EE stack.

I'll poke around and see what changed.

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
Stack:
   at Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.CompilationExtensions.GetMethod(Microsoft.CodeAnalysis.CSharp.CSharpCompilation, System.Guid, System.Reflection.Metadata.MethodDefinitionHandle)
   at Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.CompilationExtensions.GetSourceMethod(Microsoft.CodeAnalysis.CSharp.CSharpCompilation, System.Guid, Int32)
   at Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.CSharpFrameDecoder.GetMethod(Microsoft.VisualStudio.Debugger.Clr.DkmClrInstructionAddress)
   at Microsoft.CodeAnalysis.ExpressionEvaluator.FrameDecoder`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Microsoft.VisualStudio.Debugger.ComponentInterfaces.IDkmLanguageFrameDecoder.GetFrameName(Microsoft.VisualStudio.Debugger.Evaluation.DkmInspectionContext, Microsoft.VisualStudio.Debugger.DkmWorkList, Microsoft.VisualStudio.Debugger.CallStack.DkmStackWalkFrame, Microsoft.VisualStudio.Debugger.Evaluation.DkmVariableInfoFlags, Microsoft.VisualStudio.Debugger.DkmCompletionRoutine`1<Microsoft.VisualStudio.Debugger.Evaluation.DkmGetFrameNameAsyncResult>)
   at Microsoft.VisualStudio.Debugger.EntryPoint.IDkmLanguageFrameDecoder_GetFrameName(IntPtr, IntPtr, IntPtr, IntPtr, Microsoft.VisualStudio.Debugger.Evaluation.DkmVariableInfoFlags, IntPtr)
```

**Comment by latkin** [comment]
Update: since VS 2015 Preview, a re-implemented Callstack window has been introduced. Crash is occurring when loading this new callstack window. @KevinH-MS is working with debugger people to figure out fix for the nullref and/or proper logic on whether to default to old callstack window.

**Comment by latkin** [comment]
Fix for this has been checked in to Roslyn: https://github.com/dotnet/roslyn/commit/2f555565b3ca566b9aee777677808472105caa2d

Unfortunately they don't have a simple way to deploy dev bits like we do.  Fixed bits will be in the next VS CTP.  Will leave this open until then.

**Comment by forki** [comment]
Do you know an ETA?

**Comment by latkin** [comment]
ETA is whenever the hyperwebs say the next VS pre-release is coming out.

**Comment by forki** [comment]
OK. I tried with the RC. As described in https://github.com/Microsoft/visualfsharp/issues/365 I used the Paket sources set two breakpoints in build.fsx and tried to debug in interactive.

* If I highlight only a couple of lines:
   * Breakpoint works, but I don't see anything in variable windows or if I hover of existing bindings
   * So it's basically only stopping at the breakpoint without any further info
* If I highlight the whole file:
   * VS goes into some very deep sleep and doesn't respond any more

**Comment by latkin** [comment]
Since RC is out and the crash no longer occurs, I'm closing this one out.

@forki - can you open new issues for those?  FYI the debugging experience is pretty bad if you are running F# Interactive optimized (the default). I [added a warning/reminder for this](https://github.com/Microsoft/visualfsharp/pull/314#issuecomment-83827332) but somehow forgot to wire it up to "Debug in Interactive" menu item (since fixed in #355, but not in RC).  Can you double check your FSI settings?

**Comment by forki** [comment]
Thanks @latkin!

Yes the only issue is that the dialog is not coming. After manually setting the flag in the settings I can now debug FAKE scripts!

![happeing](https://cloud.githubusercontent.com/assets/57396/7420469/050a6330-ef7d-11e4-8a2e-bc4b3ff28af7.gif)

Thanks so much.
